### PR TITLE
Convert sampledata plugins to service provider

### PR DESCRIPTION
**Pull request by laoneo**
### Summary of Changes
Converts the media action plugins to service providers.

### Testing Instructions
Install sample data from the control panel of joomla.

### Actual result BEFORE applying this Pull Request
All works.

### Expected result AFTER applying this Pull Request
All works.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by ghazal** [comment]
I have tested this item :white_check_mark: successfully on 2b4d0a485a40fa9e28f78bdee69941521604be10<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40173">issues.joomla.org/tracker/joomla-cms/40173</a>.</sub>

File: plugins/sampledata/multilang/src/Extension/MultiLanguage.php
Comment (heelc29) [inline plugins/sampledata/multilang/src/Extension/MultiLanguage.php:1]: `MultiLanguage.php:897 Catching undeclared class \Joomla\Plugin\SampleData\MultiLanguage\Extension\RuntimeException (Did you mean class \RuntimeException)`
https://github.com/joomla/joomla-cms/blob/2b4d0a485a40fa9e28f78bdee69941521604be10/plugins/sampledata/multilang/src/Extension/MultiLanguage.php#L897


`MultiLanguage.php:1305 Checking instanceof against undeclared class \Joomla\Plugin\SampleData\MultiLanguage\Extension\Exception (Did you mean class \Exception)`
https://github.com/joomla/joomla-cms/blob/2b4d0a485a40fa9e28f78bdee69941521604be10/plugins/sampledata/multilang/src/Extension/MultiLanguage.php#L1305

Comment (laoneo) [reply plugins/sampledata/multilang/src/Extension/MultiLanguage.php:1]: Should be fixed with https://github.com/joomla/joomla-cms/pull/40173/commits/f305fed9eef2d789af2b6e13578c35d2c79b2c53

**Comment by heelc29** [comment]
I have tested this item :white_check_mark: successfully on 57f89a86cfa5dd466933805e88919b7c2f8c5d87<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40173">issues.joomla.org/tracker/joomla-cms/40173</a>.</sub>

**Comment by joomdonation** [comment]
I have tested this item :white_check_mark: successfully on 57f89a86cfa5dd466933805e88919b7c2f8c5d87<br /><br />Code review. Real testing by installing blog and multilingual sample data, all works well.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40173">issues.joomla.org/tracker/joomla-cms/40173</a>.</sub>

**Comment by joomdonation** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40173">issues.joomla.org/tracker/joomla-cms/40173</a>.</sub>

**Pull request merged**